### PR TITLE
Feature/py stel

### PR DESCRIPTION
**Pull request by lazersos**
This pull request improves the pySTEL libarary.

- Classes are now implemented for the various codes.
- Seperate classes now exist for dealing with input data.
- Classes wrapper fortran code for reading/writing files.
- BEAMS3D, DIAGNO, FIELDLINES, and STELLOPT input namelist routines now reside in LIBSTELL.
- A few helper utilities now exist for quickly making a few python plots.
- Included two examples for importing Blender data to render VMEC equilibria and coils data.

This closes issue #231 .

### CODE REVIEW

File: pySTEL/boozer_util.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+# Main routine
+if __name__=="__main__":
+	import sys
+	from argparse import ArgumentParser
+	import numpy as np
+	import matplotlib.pyplot as pyplot
+	from libstell.boozer import BOOZER
+	parser = ArgumentParser(description= 
+		'''Provides class for accessing boozer data also servers as a
```
Comment (leebr48) [inline pySTEL/boozer_util.py]: "also servers" -> "and also serves"

Comment (leebr48) [reply pySTEL/boozer_util.py]: This typo is repeated in at least one other pySTEL file, probably check them all.

**Review by leebr48 — APPROVED** [review]
Excellent update/addition to Stellopt.

**Comment by lazersos** [comment]
@kudav Can you check this doesn't break anything for you?

File: BEAMS3D/Sources/beams3d_grid.f90 [hunk 2]
```diff
@@ -49,16 +61,12 @@ MODULE beams3d_grid
...
                      win_raxis_fida, win_phiaxis_fida, win_zaxis_fida, win_energy_fida, win_pitch_fida, &
-                     win_beam_density, win_NEUTRONS, win_E_NEUTRONS, &
-                     nr_fida, nphi_fida, nz_fida, nenergy_fida, npitch_fida
-      REAL(rprec) :: rmin, rmax, zmin, zmax, phimin, phimax, tmin, tmax, delta_t, &
-                     vc_adapt_tol, psiedge_eq, phiedge_eq, plasma_Zmean, plasma_mass, &
-                     reff_eq, therm_factor, B_kick_min, B_kick_max, &
-      E_kick, freq_kick, t_fida, rho_fullorbit, rho_help,&
-      s_max,s_max_te, s_max_ne,s_max_zeff,s_max_ti, s_max_pot
+                     win_beam_density, win_NEUTRONS, win_E_NEUTRONS 
+      REAL :: emin_fida, pimin_fida
```
Comment (jonathanschilling) [inline BEAMS3D/Sources/beams3d_grid.f90]: Is it indented to not have `rprec` specified here explicitly?

Comment (lazersos) [reply BEAMS3D/Sources/beams3d_grid.f90]: @kudav can you comment on this.

Comment (kudav) [reply BEAMS3D/Sources/beams3d_grid.f90]: Not intentional, I just pushed a fix.

File: pySTEL/libstell/fieldlines.py [hunk 3]
```diff
@@ -1,75 +1,209 @@
...
+		iota_err : ndarray
+			Error in rotational transform
+		"""
+		import numpy as np
+		x = self.R_lines
+		y = self.Z_lines
+		for i in range(self.nsteps):
+			x[i,:] = x[i,:] - x[i,0]
+			y[i,:] = y[i,:] - y[i,0]
+		theta = np.arctan2(y,x)
```
Comment (jons-pf) [inline pySTEL/libstell/fieldlines.py]: line 115 is a duplicate of line 116?

**Review by jons-pf — APPROVED** [review]
(just noticed that I appear now in the list of reviewers and don't want to block the merge in case GitHub would wait for me)
Looks good to me.

**Review by jonathanschilling — APPROVED** [review]
(just noticed that I appear now in the list of reviewers and don't want to block the merge in case GitHub would wait for me)
Looks good to me.

**Pull request merged**